canonical-web-and-design/maas-ui
# Add checkboxes to KVM list table

## Done

- Added checkboxes to KVM list table
- Added `pod.selected` to state (+ relevant actions, reducers, selectors)

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/kvm
- Check in Redux DevTools that there is a `pod.selected` value of `[]`;.
- Select some checkboxes and check that the header count updates accordingly, and `SET_SELECTED_PODS` actions are dispatched with the correct pod ids.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2023

## Screenshot
![Screenshot_2020-06-22 KVM bolla MAAS](https://user-images.githubusercontent.com/25733845/85243775-12aa7600-b486-11ea-9413-18a96cd7e3fc.png)
